modules/client/src/main/scala/higherkindness/compendium/CompendiumClient.scala

```diff
@@ -16,99 +16,104 @@
 
-      override def storeProtocol(identifier: String, protocol: Protocol): IO[Int] = {
-        val request =
+      override def storeProtocol(identifier: String, protocol: Protocol): F[Int] = {
+        val request: Free[HttpF, HttpResponse] =
           Hammock.request(Method.POST, uri"$baseUrl/v0/protocol/$identifier", Map(), Some(protocol))
 
         for {
-          status <- request.map(_.status).exec[IO]
-          _ <- status match {
-            case Status.Created => IO.unit
-            case Status.OK      => IO.unit
-            case Status.BadRequest =>
-              asError(request, SchemaError)
-            case Status.InternalServerError =>
-              IO.raiseError(UnknownError(s"Error in compendium server"))
-            case _ =>
-              IO.raiseError(UnknownError(s"Unknown error with status code $status"))
+          status <- request.map(_.status).exec[F]
+          _ <- {
+            val statusToF: F[Unit] =
+              status match {
+                case Status.Created => Sync[F].unit
+                case Status.OK      => Sync[F].unit
+                case Status.BadRequest =>
+                  asError(request, SchemaError)
+                case Status.InternalServerError =>
+                  Sync[F].raiseError(UnknownError(s"Error in compendium server"))
+                case _ =>
+                  Sync[F].raiseError(UnknownError(s"Unknown error with status code $status"))
+              }
+            statusToF
           }
         } yield status.code
       }
 
       override def retrieveProtocol(
           identifier: String,
-          version: Option[Int]): IO[Option[Protocol]] = {
+          version: Option[Int]): F[Option[Protocol]] = {
         val versionParam = version.fold("")(v => s"?$versionParamName=${v.show}")
         val uri          = uri"$baseUrl/v0/protocol/$identifier$versionParam"
 
         val request: Free[HttpF, HttpResponse] = Hammock.request(Method.GET, uri, Map())
 
         for {
-          status <- request.map(_.status).exec[IO]
-          out <- status match {
-            case Status.OK       => request.as[Protocol].map(Some(_)).exec[IO]
-            case Status.NotFound => IO(None)
-            case Status.InternalServerError =>
-              IO.raiseError(UnknownError(s"Error in compendium server"))
-            case _ =>
-              IO.raiseError(UnknownError(s"Unknown error with status code $status"))
+          status <- request.map(_.status).exec[F]
+          out <- {
+            val statusToF: F[Option[Protocol]] = status match {
+              case Status.OK       => request.as[Protocol].map(Option(_)).exec[F]
+              case Status.NotFound => Sync[F].pure(None)
+              case Status.InternalServerError =>
+                Sync[F].raiseError(UnknownError(s"Error in compendium server"))
+              case _ =>
+                Sync[F].raiseError(UnknownError(s"Unknown error with status code $status"))
+            }
+            statusToF
```

Comment:
Just out of curiosity, why creating `statusToF` to return it later?

Reply:
Ah, good question, I initially tried without the `val` and it seems the Scala compiler can't interpret the case statement as total on the right side of the for expression without a little bit of type help. I'm open to ideas here, as I would have preferred not to create the value only to return it.

Reply:
Nice! Thank you @rlmark !!